From 0c77c24ec992babe713fd8ba0ed35c81a684f97a Mon Sep 17 00:00:00 2001
From: marshmallow <marshycity@gmail.com>
Date: Sun, 30 Jun 2024 20:52:09 +1000
Subject: [PATCH] NIX_ABS_CONFIG

---
 src/nvim/os/stdpaths.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/nvim/os/stdpaths.c b/src/nvim/os/stdpaths.c
index e5bdd56fe..9340e4320 100644
--- a/src/nvim/os/stdpaths.c
+++ b/src/nvim/os/stdpaths.c
@@ -178,6 +178,12 @@ char *stdpaths_get_xdg_var(const XDGVarType idx)
     ret = xdg_remove_duplicate(ret, ENV_SEPSTR);
   }
 
+  char *nix = os_getenv("NIX_ABS_CONFIG");
+
+  if (idx == kXDGConfigHome && nix != NULL) {
+    ret = xstrdup(nix);
+  }
+
   return ret;
 }
 
-- 
2.44.0

